### PR TITLE
Wip/hughsie/vendor

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: src/fu-udev-device.c [hunk 1]
```diff
@@ -93,9 +92,17 @@ fu_udev_device_probe (FuDevice *device, GError **error)
...
+	priv->model = fu_udev_device_get_sysfs_attr_as_uint64 (priv->udev_device, "device");
+	priv->revision = fu_udev_device_get_sysfs_attr_as_uint64 (priv->udev_device, "revision");
+
+	/* fallback to the parent */
+	if (priv->vendor == 0x0 && priv->model == 0x0 && priv->revision == 0x0) {
+		g_autoptr(GUdevDevice) parent = g_udev_device_get_parent (priv->udev_device);
+		priv->vendor = fu_udev_device_get_sysfs_attr_as_uint64 (parent, "vendor");
+		priv->model = fu_udev_device_get_sysfs_attr_as_uint64 (parent, "device");
+		priv->revision = fu_udev_device_get_sysfs_attr_as_uint64 (parent, "revision");
+	}
```
[the remark this answers is not in the document]
Comment (superm1) [reply src/fu-udev-device.c]: (I know it helps the NVME case though; just wondering if there is a missing safety check)

**Comment by superm1** [comment]
Given I've had flash success, thoughts on removing https://github.com/hughsie/fwupd/blob/master/plugins/nvme/fu-plugin-nvme.c#L45 ?

**Comment by hughsie** [comment]
> Given I've had flash success

Yup, we should remove the need for --force -- can you add it to this PR pls.

File: src/fu-udev-device.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (hughsie) [reply src/fu-udev-device.c]: > Not sure falling back parent will always be valid

I can't think of an example of where it wouldn't be; if the GUdevDevice has no class then it's probably a "virtual" device layered on top of something like USB or PCI. I guess maybe we could achieve the same thing using `g_udev_device_get_parent_with_subsystem("pci")` but that hardcodes the subsystem. Or maybe better still check for `class` when getting the vendor and go down the tree until we find it.

[the remark this answers is not in the document]
Comment (superm1) [reply src/fu-udev-device.c]: I think that check is exactly what makes sense.  Compare that `g_udev_device_get_subsystem` matches for both parent and child.

[the remark this answers is not in the document]
Comment (hughsie) [reply src/fu-udev-device.c]: I tried this. Going down all the parents until we find `class` means that for actual devices that are classless go down to the USB hub, and so they get all identified as Intel. For instance, hidraw.

[the remark this answers is not in the document]
Comment (superm1) [reply src/fu-udev-device.c]: Ah I see.  Then maybe class makes sense indeed.

**Pull request merged**